src/daft-core/src/datatypes/dtype.rs

```diff
@@ -334,6 +334,43 @@ impl DataType {
         }
     }
 
+    pub fn estimate_size_bytes(&self) -> Option<f64> {
```

Comment:
Seems like the only case that we return `None` for is `_`

Could we just have it return `f64` and when we add a new physical type, just add a new branch here? Seems like that'd be better behavior than naively returning `None`, which translates to an estimate of `0.` in upstream code